lib/ansible/modules/windows/win_timezone.ps1

```diff
@@ -19,28 +19,32 @@
 # WANT_JSON
 # POWERSHELL_COMMON
 
-$params = Parse-Args $args;
-
-$result = New-Object psobject @{
-    win_timezone = New-Object psobject
+$result = @{
     changed = $false
+    win_timezone = @{}
 }
 
-$timezone = Get-Attr -obj $params -name timezone -failifempty $true -resultobj $result
+$params = Parse-Args $args -supports_check_mode $true
+$check_mode = Get-AnsibleParam -obj $params -name "_ansible_check_mode" -type "bool" -default $false
+
+$timezone = Get-AnsibleParam -obj $params -name "timezone" -failifempty $true -resultobj $result
```

Comment:
To be honest, the -resultobj does not seem to do anything except be used for error-handling within the AnsibleParam function. So I would remove them myself, but I don't think I am smarter than the person who added it, so I restricted myself to the stuff I do know and are safe to change.